### ISSUE TITLE
Char combinations

### ISSUE THREAD

**Issue by iegik** [andyhu/node-transliteration#15]
```
            {"и", "i"},
            {"й", "y"},
            {"е", "e"},
```
but

```
            {"ое", "oye"},
            {"aе", "aye"},
            {"уе", "uye"},
            {"ие", "iye"},
            {"эе", "eye"},
            {"ье", "ye"},
            {"ъа", "yа"},
            {"ьы", "y"},
            {"ьи", "yi"},
            {"ьо", "yo"},
            {"ьу", "yu"},
            {"ый", "y"},
            {"ий", "iy"},
```
or
```
{{"ку", "кью"}, {"q"}}
```

References:

https://github.com/octo47/t13n/blob/master/src/main/java/ru/creditnet/util/t13n/lang/Ru.java
https://github.com/gioele/t13n-fixtures-sanskrit

**Comment by andyhu** [comment]
Thanks for bringing this topic up.
I actually thought about using an in-memory database as the data source (also support persistent data) and probably provide an online repository to let users to choose a downloadable char map / charset map so users could contribute and maintain their own char maps (probably transliteration.io?), they could also share it. Once downloaded the data is stored locally. I'm thinking to use the LokiJS as an embedded db which has very good performance and can be used in both node.js and browser. If we decided to go this way, it's sure that the code must be refactored again. What do you think?

**Comment by iegik** [comment]
Iou can provide your "database solution" in additional way, but keep char mappings separatly

**Comment by iegik** [comment]
BTW, better keep pairs in Map type - key-value pairs (2-element Arrays):

```
Cyrilic.iso9a.ru = [
// combinations
...
// single chars
  ["а", "a"],
...
  ["я", "ya"]
]
```

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

And, please, do not uglify sources! Keep each pair in one row - easy to read and convert to other formats (programming languages).

```
Cyrilic.iso9a.ru = {
// combinations
...
// single chars
  "а" => "a",
...
  "я" => "ya"
}
```

**Comment by andyhu** [comment]
Hi @iegik, thanks for the suggestions. However, before deciding to do it in this way, there are still two issues I'm thinking about:
1. Where can we get the data from?
2. if it's not a full charmap, it's more verbose and will need around 4x space to store those data. the javascript file will be larger than 1MB which is not quite practical for the usage in the browser.

**Comment by andyhu** [comment]
Character combinations replacement feature is implemented in v1.0.0+. Closing this this issue for now
